### PR TITLE
Framework: Add CDash reporting to AT2 builds

### PR DESCRIPTION
**Pull request by jmlapre**
See https://sems-atlassian-son.sandia.gov/jira/browse/TRILFRAME-627

@trilinos/framework 

## Motivation

Attempt to use PullRequestLinuxDriverTest.py to drive in-container PR testing (this will in-turn use SimpleTestingRefactor, which will submit to CDash).

Keep in mind the verbosity of the scripts, and reduce if it is thought prudent.

### CODE REVIEW

File: .github/workflows/AT2.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
           bash -l -c "module list"
           printenv PATH
       - name: Cancel Previous Runs
-        uses: styfle/cancel-workflow-action@85880fa0301c86cca9da44039ee3bb12d3bedbfa # 0.12.1
+        uses: styfle/cancel-workflow-action@0.12.1
```
Comment (sebrowne) [inline .github/workflows/AT2.yml:35]: Leave these at the SHA1s for security reasons.

File: .github/workflows/AT2.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
           mkdir -p /home/Trilinos/src/Trilinos
           mkdir -p /home/Trilinos/build
       - name: Clone trilinos
-        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+        uses: actions/checkout@v4.1.2
```
Comment (sebrowne) [inline .github/workflows/AT2.yml:44]: Same as above

File: .github/workflows/AT2.yml [hunk 3]
```diff
@@ -56,24 +56,34 @@ jobs:
...
+          sed -i "/\[rhel8\]/a `cat /etc/hostname`" /home/runner/_work/Trilinos/Trilinos/packages/framework/ini-files/supported-systems.ini
+          printf "\n\n\n"
+
+          python ${GITHUB_WORKSPACE}/packages/framework/pr_tools/PullRequestLinuxDriverTest.py \
+            --source-repo-url ${GITHUB_WORKSPACE} \
+            --target-repo-url ${GITHUB_WORKSPACE} \
+            --target-branch-name ${{ github.event.pull_request.base.ref }} \
+            --pullrequest-build-name PR-${{ github.event.number }} \
+            --genconfig-build-name rhel8_gcc-openmpi_debug_shared_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables \
+            --pullrequest-number 0 \
```
Comment (sebrowne) [inline .github/workflows/AT2.yml:75]: Can we use `${{ github.event.number }}` here as well?

File: .github/workflows/AT2.yml [hunk 4]
```diff
@@ -90,7 +100,7 @@ jobs:
           bash -l -c "module list"
           printenv PATH
       - name: Cancel Previous Runs
-        uses: styfle/cancel-workflow-action@85880fa0301c86cca9da44039ee3bb12d3bedbfa # 0.12.1
+        uses: styfle/cancel-workflow-action@0.12.1
```
Comment (sebrowne) [inline .github/workflows/AT2.yml:103]: Same as other comment about sha1s

File: .github/workflows/AT2.yml [hunk 5]
```diff
@@ -99,7 +109,7 @@ jobs:
           mkdir -p /home/Trilinos/src/Trilinos
           mkdir -p /home/Trilinos/build
       - name: Clone trilinos
-        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+        uses: actions/checkout@v4.1.2
```
Comment (sebrowne) [inline .github/workflows/AT2.yml:112]: Same as other comment about sha1s

**Review by sebrowne — CHANGES_REQUESTED** [review]
Requesting the versions of the actions used be restored to the current SHA1s

**Review by sebrowne — APPROVED** [review]

**Review by achauphan — APPROVED** [review]

**Pull request merged**